contracts/FastBreak.cdc

```diff
@@ -0,0 +1,682 @@
+
+    /// Fast Break Daemon game oracle implementation
+    ///
+    pub resource FastBreakDaemon: GameOracle {
+
+        /// Create a Fast Break Run
+        ///
+        pub fun createFastBreakRun(id: String, name: String, runStart: UInt64, runEnd: UInt64, fatigueModeOn: Bool) {
+            let fastBreakRun = FastBreak.FastBreakRun(
+                id: id,
+                name: name,
+                runStart: runStart,
+                runEnd: runEnd,
+                fatigueModeOn: fatigueModeOn
+            )
+            FastBreak.fastBreakRunByID[fastBreakRun.id] = fastBreakRun
+            emit FastBreakRunCreated(
+                id: fastBreakRun.id,
+                name: fastBreakRun.name,
+                runStart: fastBreakRun.runStart,
+                runEnd: fastBreakRun.runEnd,
+                fatigueModeOn: fastBreakRun.fatigueModeOn
+            )
+        }
+
+        /// Update the status of a Fast Break Run
+        ///
+        pub fun updateFastBreakRunStatus(id: String, status: String) {
+            let fastBreakRun = (&FastBreak.fastBreakRunByID[id] as &FastBreak.FastBreakRun?)!
+
+            fastBreakRun.updateStatus(status: status)
+            emit FastBreakRunStatusChange(id: fastBreakRun.id, newStatus: fastBreakRun.status)
+        }
+
+        /// Create a game of Fast Break
+        ///
+        pub fun createFastBreakGame(
+            id: String,
+            name: String,
+            fastBreakRunID: String,
+            isPublic: Bool,
+            submissionDeadline: UInt64,
+            numPlayers: UInt64
+        ) {
+            let fastBreakGame: FastBreak.FastBreakGame = FastBreak.FastBreakGame(
+                id: id,
+                name: name,
+                fastBreakRunID: fastBreakRunID,
+                isPublic: isPublic,
+                submissionDeadline: submissionDeadline,
+                numPlayers: numPlayers
+            )
+            FastBreak.fastBreakGameByID[fastBreakGame.id] = fastBreakGame
+            emit FastBreakGameCreated(
+                id: fastBreakGame.id,
+                name: fastBreakGame.name,
+                fastBreakRunID: fastBreakGame.fastBreakRunID,
+                isPublic: fastBreakGame.isPublic,
+                submissionDeadline: fastBreakGame.submissionDeadline,
+                numPlayers: fastBreakGame.numPlayers
+            )
+        }
+
+        /// Add a Fast Break Statistic to a game of Fast Break during game creation
+        ///
+        pub fun addStatToFastBreakGame(fastBreakGameID: String, name: String, type: String, valueNeeded: UInt64) {
+            let fastBreakGame: &FastBreak.FastBreakGame = (&FastBreak.fastBreakGameByID[fastBreakGameID] as &FastBreak.FastBreakGame?)!
+            let fastBreakStat : FastBreak.FastBreakStat = FastBreak.FastBreakStat(
+                name: name,
+                type: type,
+                valueNeeded: valueNeeded
+            )
+
+            fastBreakGame.addStat(stat: fastBreakStat)
```

Comment:
Do we need to leave open the possibility of deleting or modifying a stat in the event of an incorrect stat getting added?

Reply:
Currently this is a no in the off-chain game because it would strand wallets who already submitted to the fast break.

One change I want to make in the short term is to allow edits until the Fast Break is in an open state. But this may not ever get prioritized, so for the moment, you cannot edit stats once a fast break has been created. Must do so carefully like when we make sets & series.